src/plone/restapi/indexers.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+from plone.restapi.behaviors import IBlocks
+from plone.indexer.decorator import indexer
+from plone.app.contenttypes.indexers import SearchableText
+import six
+
+
+def _extract_text(block):
+    result = ""
+    for paragraph in block.get("text").get("blocks"):
+        text = paragraph["text"]
+        if six.PY2:
+            if isinstance(text, six.text_type):
+                text = text.encode("utf-8", "replace")
+            if text:
+                result = " ".join((result, text))
+        else:
+            result = " ".join((result, text))
```

Comment:
```
from Products.CMPPlone.utils import safe_nativestring
" ".join(result, safe_nativestring(text))
```
does the same with lot less lines.
Making a single `" ".join` operation at the end would also be more efficient.
Something like:
```
return " ".join(safe_nativestring(paragraph["text"]) for paragraph in block.get("text").get("blocks"))
```

Reply:
@ale-rt fixed.

Reply:
@ale-rt I had to revert my fix. plone.restapi supports Plone 4.3 which does not have safe_nativestring. :(